src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -142078,5 +142121,658 @@ static T GetValue2<T>(Collection c, object key)
+            var model = comp.GetSemanticModel(tree, ignoreAccessibility: false);
+            var declarators = tree.GetRoot().DescendantNodes().OfType<EqualsValueClauseSyntax>().ToArray();
+            Assert.Equal("field", declarators[0].Value.ToString());
+            var field1 = model.GetSymbolInfo(declarators[0].Value).Symbol;
+            Assert.Equal("C<T>.field", declarators[1].Value.ToString());
+            var field2 = model.GetSymbolInfo(declarators[1].Value).Symbol;
+            Assert.Equal("C<T>.field", declarators[2].Value.ToString());
+            var field3 = model.GetSymbolInfo(declarators[2].Value).Symbol;
+
+            Assert.True(field2.Equals(field3, SymbolEqualityComparer.Default));
+            Assert.False(field2.Equals(field3, SymbolEqualityComparer.IncludeNullability));
+            Assert.True(field3.Equals(field2, SymbolEqualityComparer.Default));
+            Assert.False(field3.Equals(field2, SymbolEqualityComparer.IncludeNullability));
+
+            Assert.True(field1.Equals(field2, SymbolEqualityComparer.Default));
+            Assert.False(field1.Equals(field2, SymbolEqualityComparer.IncludeNullability));
+            Assert.True(field2.Equals(field1, SymbolEqualityComparer.Default));
+            Assert.False(field2.Equals(field1, SymbolEqualityComparer.IncludeNullability));
+
+            Assert.True(field1.Equals(field3, SymbolEqualityComparer.Default));
+            Assert.True(field1.Equals(field3, SymbolEqualityComparer.IncludeNullability));
+            Assert.True(field3.Equals(field1, SymbolEqualityComparer.Default));
+            Assert.True(field3.Equals(field1, SymbolEqualityComparer.IncludeNullability));
+        }
+
+        [Fact]
+        public void IgnoredNullability_MethodSymbol()
+        {
+            var src = @"
+#nullable enable
+
+public class C
+{
+    public void M<T>(out T x)
+    {
+        M<T>(out x);
+#nullable disable
+        M<T>(out x);
+    }
+}
+";
+            var comp = CreateCompilation(src);
+            comp.VerifyDiagnostics();
+
+            var tree = comp.SyntaxTrees.Single();
+            var model = comp.GetSemanticModel(tree, ignoreAccessibility: false);
+            var method1 = model.GetDeclaredSymbol(tree.GetRoot().DescendantNodes().OfType<MethodDeclarationSyntax>().Single());
+            Assert.True(method1.IsDefinition);
+            var invocations = tree.GetRoot().DescendantNodes().OfType<InvocationExpressionSyntax>().ToArray();
+            Assert.Equal("M<T>(out x)", invocations[0].ToString());
+            var method2 = model.GetSymbolInfo(invocations[0]).Symbol;
+            Assert.False(method2.IsDefinition);
+            Assert.Equal("M<T>(out x)", invocations[1].ToString());
+            var method3 = model.GetSymbolInfo(invocations[1]).Symbol;
+            Assert.True(method3.IsDefinition);
+
+            // Note: definitions and subsituted symbols are not equal even when ignoring nullability
```

Comment:
>// Note: definitions and subsituted symbols are not equal even when ignoring nullability [](start = 12, length = 88)

@alekseyts @cston FYI This behavior seems wrong, but I didn't file an issue for this because from our discussion, trying to adjust this seems very risky (likely performance impact and other unknowns).

Reply:
> This behavior seems wrong, but I didn't file an issue for this because from our discussion, trying to adjust this seems very risky (likely performance impact and other unknowns). 

I do not remember us discussing this, unless this somehow involves "Ignored" annotation. It doesn't look like it does. I think we should fix this. Not necessary in this PR, unless the behavior is related to the changes in it.


---
In reply to: [536369020](https://github.com/dotnet/roslyn/pull/49723#discussion_r536369020) [](ancestors = 536369020)

Reply:
Okay. Thanks for the clarification. Filed https://github.com/dotnet/roslyn/issues/49798

Right, we discussed this for comparisons dealing with Ignored annotation, but we have the same problem for existing annotations (shortcuts we take when one of the symbols is a definition).